lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -84,95 +105,83 @@ UsdSceneItem::Ptr UsdUndoRenameCommand::renamedItem() const
 
 bool UsdUndoRenameCommand::renameRedo()
 {
-    const UsdPrim& prim = _stage->GetPrimAtPath(_ufeSrcItem->prim().GetPath());
-
-    auto primSpec = MayaUsdUtils::getPrimSpecAtEditTarget(prim);
-    if(!primSpec) {
-        return false;
-    }
-
-    // these two lines MUST be called before the set name
-    // _stage->GetDefaultPrim() and prim after the rename can be invalid.
-    auto primPath = prim.GetPath();
+    // get the stage's default prim path
     auto defaultPrimPath = _stage->GetDefaultPrim().GetPath();
 
-    // handle unique name for _newName
-    TfToken::HashSet childrenNames;
-    for (auto child : prim.GetParent().GetChildren()){
-        childrenNames.insert(child.GetName());
-    }
-    if (childrenNames.find(TfToken(_newName)) != childrenNames.end()){
-        _newName = uniqueName(childrenNames, _newName);
-    }
+    // 1- open a changeblock to delay sending notifications.
+    // 2- update the Internal References paths (if any) first
+    // 3- set the new name 
+    {
+        SdfChangeBlock changeBlock;
 
-    // names are not allowed to start to digit numbers
-    if(std::isdigit(_newName.at(0))){
-        _newName = prim.GetName();
-    }
+        const UsdPrim& prim = _stage->GetPrimAtPath(_ufeSrcItem->prim().GetPath());
 
-    // all special characters are replaced with `_`
-    const std::string specialChars{"~!@#$%^&*()-=+,.?`':{}|<>[]/ "};
-    std::replace_if(_newName.begin(), _newName.end(), [&](auto c){
-        return std::string::npos != specialChars.find(c);
-    }, '_');
+        auto newSceneItem = createSiblingSceneItem(_ufeSrcItem->path(), _newName);
```

Comment:
Instead of doing the string manipulation, can't we just dynamic_cast to a UsdSceneItem and get the new prim's path, much like on (new) line 117?

Reply:
Hey @ppt-adsk let's have a chat offline tomorrow. I am not sure if the dynamic_cast would work here but maybe I am missing something obvious.